src/main/webapp/add_event_page/script.js

```diff
@@ -1,22 +1,116 @@
 let map;
-function initMap(){
+let school_id = 0;
+let school_position;
+let initial_lat;
+let initial_lng;
+let currentLatitude;
+let currentLongitude;
+
+window.onload = function() {
+    this.getSchool();
+    this.initMap();
+    this.document.getElementById('back').setAttribute('onclick', `location.href = '../directory-page/directory.html?school_id=${school_id}'`);
+    this.document.getElementById('form').setAttribute('action', `../directory-page/directory.html?school_id=${school_id}`);
+}
+
+function getSchool(){
+    new_search = new URLSearchParams(window.location.search);
+    school_id = new_search.get("school_id");
+}
+
+async function getPos(){
+    response = await fetch(`/school?school_id=${school_id}`)
+    school_info = await response.json();
+    school_position = school_info.position;
+    initial_lat = parseFloat(school_position.latitude);
+    initial_lng = parseFloat(school_position.longitude);
+    currentLatitude = initial_lat;
+    currentLongitude = initial_lng;
+}
+
+async function initMap(){
+    await this.getPos();
     map = new google.maps.Map(document.getElementById("map"),{
-        center: {lat:32.8811, lng:-117.2376},
+        center: {lat:initial_lat, lng:initial_lng},
         zoom: 16,
     });
 
     var marker = new google.maps.Marker({
-        position: {lat:32.8811, lng:-117.2376},
+        position: {lat:initial_lat, lng:initial_lng},
         map: map,
         draggable: true,
     });
 
+    const input = this.document.getElementById("place-input");
+    const searchBox = new google.maps.places.SearchBox(input);
+
+    google.maps.event.addListener(searchBox,'places_changed',function(){
+        var places = searchBox.getPlaces();
+        var bounds = new google.maps.LatLngBounds();
+        var i,place;
+        for(i=0; place=places[i]; i++){
+          bounds.extend(place.geometry.location);
+          marker.setPosition(place.geometry.location);
+          currentLatitude = marker.latLng.lat();
+          currentLongitude = marker.latLng.lng();
+        }
+        map.fitBounds(bounds);
+        map.setZoom(15);
+    });
+
     google.maps.event.addListener(marker, 'dragend', function(marker){
         var latLng = marker.latLng; 
         currentLatitude = latLng.lat();
         currentLongitude = latLng.lng();
-        jQ("#latitude").val(currentLatitude);
-        jQ("#longitude").val(currentLongitude);
     }); 
 }
-window.initMap = initMap;
\ No newline at end of file
+window.initMap = initMap;
+
+function checkType(){
+    if (this.document.getElementById("event_type").value == "Study") {
+        this.document.getElementById("subject1").style.display = '';
+        this.document.getElementById("subject2").style.display = '';
+    } else {
+        this.document.getElementById("subject1").style.display = 'none';
+        this.document.getElementById("subject2").style.display = 'none';
+    }
+}
+
+function checkMin(){
```

Comment:
Thanks for adding this! I would call this in window.onload so the date options are already limited before the user has picked one.